### ISSUE TITLE
2.4.39: Breaks IDE without kotlin? plugin (pycharm, clion, etc...)

### ISSUE THREAD

**Issue by topia** [snyk/snyk-intellij-plugin#403]
Editor completely broken
```
2022-07-20 22:27:50,739 [  34710] SEVERE - #c.i.i.s.i.StartupManagerImpl - PyCharm 2022.2 EAP  Build #PY-222.3345.40
2022-07-20 22:27:50,739 [  34710] SEVERE - #c.i.i.s.i.StartupManagerImpl - JDK: 17.0.3; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o.
2022-07-20 22:27:50,739 [  34710] SEVERE - #c.i.i.s.i.StartupManagerImpl - OS: Windows 11
2022-07-20 22:27:50,739 [  34710] SEVERE - #c.i.i.s.i.StartupManagerImpl - Plugin to blame: Snyk Security - Code, Open Source, Container, IaC Configurations version: 2.4.39
2022-07-20 22:30:00,386 [ 164357] SEVERE - #c.i.i.p.PluginManager - Cannot create class io.snyk.plugin.services.SnykApplicationSettingsStateService (classloader=PluginClassLoader(plugin=PluginDescriptor(name=Snyk Security - Code, Open Source, Container, IaC Configurations, id=io.snyk.snyk-intellij-plugin, descriptorPath=plugin.xml, path=~\AppData\Local\JetBrains\Toolbox\apps\PyCharm-P\ch-0\222.3345.40.plugins\snyk-intellij-plugin, version=2.4.39, package=null, isBundled=false), packagePrefix=null, instanceId=140, state=active))
com.intellij.diagnostic.PluginException: Cannot create class io.snyk.plugin.services.SnykApplicationSettingsStateService (classloader=PluginClassLoader(plugin=PluginDescriptor(name=Snyk Security - Code, Open Source, Container, IaC Configurations, id=io.snyk.snyk-intellij-plugin, descriptorPath=plugin.xml, path=~\AppData\Local\JetBrains\Toolbox\apps\PyCharm-P\ch-0\222.3345.40.plugins\snyk-intellij-plugin, version=2.4.39, package=null, isBundled=false), packagePrefix=null, instanceId=140, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:883)
	at com.intellij.serviceContainer.ComponentManagerImpl.createLightService(ComponentManagerImpl.kt:823)
	at com.intellij.serviceContainer.ComponentManagerImpl.getOrCreateLightService(ComponentManagerImpl.kt:669)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:621)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:583)
	at com.intellij.openapi.client.ClientAwareComponentManager.getFromSelfOrCurrentSession(ClientAwareComponentManager.kt:37)
	at com.intellij.openapi.client.ClientAwareComponentManager.getService(ClientAwareComponentManager.kt:22)
	at io.snyk.plugin.UtilsKt.pluginSettings(Utils.kt:390)
	at snyk.advisor.AdvisorScoreProvider.getLineExtensions(AdvisorScoreProvider.kt:40)
	at snyk.advisor.AdvisorEditorFactoryListener$editorCreated$1.invoke(AdvisorEditorFactoryListener.kt:36)
	at snyk.advisor.AdvisorEditorFactoryListener$editorCreated$1.invoke(AdvisorEditorFactoryListener.kt:9)
	at snyk.advisor.AdvisorEditorFactoryListener$sam$java_util_function_IntFunction$0.apply(AdvisorEditorFactoryListener.kt)
	at com.intellij.openapi.editor.impl.EditorImpl.processLineExtensions(EditorImpl.java:770)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.collectExtensions(EditorPainter.java:857)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session$1.paintAfterLineEnd(EditorPainter.java:443)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paintLineFragments(EditorPainter.java:1550)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paintBackground(EditorPainter.java:356)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paint(EditorPainter.java:184)
	at com.intellij.openapi.editor.impl.view.EditorPainter.paint(EditorPainter.java:71)
	at com.intellij.openapi.editor.impl.view.EditorView.paint(EditorView.java:287)
	at com.intellij.openapi.editor.impl.EditorImpl.paint(EditorImpl.java:1965)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.paintComponent(EditorComponentImpl.java:253)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1122)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.paint(EditorComponentImpl.java:124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JViewport.paint(JViewport.java:736)
	at com.intellij.ui.components.JBViewport.paint(JBViewport.java:235)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at com.intellij.ui.tabs.impl.JBTabsImpl.paintChildren(JBTabsImpl.java:583)
	at com.intellij.openapi.fileEditor.impl.EditorTabbedContainer$EditorTabs.paintChildren(EditorTabbedContainer.java:724)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1131)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5319)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBufferedImpl(RepaintManager.java:1668)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1643)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1580)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1347)
	at java.desktop/javax.swing.JComponent._paintImmediately(JComponent.java:5267)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5077)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:882)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:865)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:865)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:838)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:787)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1909)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:479)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/konan/file/File
	at io.snyk.plugin.services.SnykApplicationSettingsStateService.<init>(SnykApplicationSettingsStateService.kt:30)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:844)
	... 88 more
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.konan.file.File PluginClassLoader(plugin=PluginDescriptor(name=Snyk Security - Code, Open Source, Container, IaC Configurations, id=io.snyk.snyk-intellij-plugin, descriptorPath=plugin.xml, path=~\AppData\Local\JetBrains\Toolbox\apps\PyCharm-P\ch-0\222.3345.40.plugins\snyk-intellij-plugin, version=2.4.39, package=null, isBundled=false), packagePrefix=null, instanceId=140, state=active)
	... 90 more
```

sample screenshot:
![image](https://user-images.githubusercontent.com/13070/180021434-f263f13f-71ce-4d64-a87a-90054f57ed43.png)

**Comment by topia** [comment]
I suspect this line - import wrong class: https://github.com/snyk/snyk-intellij-plugin/commit/efa9acb4d44df7a0ec1177875a8179b15bddecea#diff-c27dda896e71ad44fa605baf518516de295664a15ee99dbc46af68f5e2303804R15

**Comment by gauravcanon** [comment]
Have the same issue in PHPStrom and PyCharm currently the only option is to disable

**Comment by topia** [comment]
I disabled it too, but for someone who wants to use the plugin asap, may downloading an older one works: https://plugins.jetbrains.com/plugin/10972-snyk-security--code-open-source-container-iac-configurations/versions/stable/194719

**Comment by wolfie82** [comment]
I opened a Snyk support ticket about this issue yesterday (28462).  My issue is similar, except that I'm using Rider.  Upon project load, the IDE just grinds to a halt with some UI functionality still expressed, however its severely broken and unusable.  Manually downloading the pervious version and installing resolved my issue.

**Comment by bastiandoetsch** [comment]
I'm looking into it :). Thanks for reporting!

**Comment by bastiandoetsch** [comment]
@asaf92 fyi

**Comment by bastiandoetsch** [comment]
@asaf92 Please verify the fix in Jetbrains Rider. 
@wolfie82 & @topia We will try to get this published the fix as quickly as possible on Jetbrains marketplace, submitting it hopefully on Jul 21st. Jetbrains' review can take up to two days unfortunately, though.

**Comment by francoisstark400** [comment]
Also affected me - same issue, cotlin libraries not found.  
I installed the older plugin, 2.4.38 and the problem goes away.  Ubuntu phpstorm client.

**Comment by bastiandoetsch** [comment]
Jetbrains has finished the review and now released version 2.4.40 on its marketplace

**Comment by bastiandoetsch** [comment]
Jetbrains has finished the review and now released version 2.4.40 on its marketplace
